semantic_model_generator/validate/context_length.py

```diff
@@ -1,13 +1,66 @@
-from typing import TypeVar
+from typing import Any, TypeVar
 
 from google.protobuf.message import Message
 from loguru import logger
 
 from semantic_model_generator.data_processing.proto_utils import proto_to_yaml
 
 ProtoMsg = TypeVar("ProtoMsg", bound=Message)
-_MODEL_CONTEXT_LENGTH_TOKENS = 6500  # We use 6.5k, with 1.2k for instructions, so that we can reserve 500 for response tokens (average is 300).
-_MODEL_CONTEXT_INSTR_TOKEN = 20  # buffer for instr tokens
+
+# Max total tokens is 8200.
+# We reserve 500 tokens for response (average response is 300 tokens).
+# So the prompt token limit is 7700.
+_TOTAL_PROMPT_TOKEN_LIMIT = 7700
+_BASE_INSTRUCTION_TOKEN_LENGTH = 1220
```

Comment:
does this count in max 3 sample value for each column already?

Reply:
I'm assuming we want to save 1220 tokens for all the instructions around the semantic model.
So the semantic model will get about (7700 -1220) tokens, with some more discounting for retrieved literals.

The max 3 sample values for each column will be counted as part of the semantic model's tokens.

Reply:
Got it, thanks!

Can you add the comment to what `_BASE_INSTRUCTION_TOKEN_LENGTH` includes?

Reply:
done